src/kaleidoscope/hardware/kbdfans/KBD4x.h

```diff
@@ -19,37 +19,36 @@
 
 #ifdef ARDUINO_AVR_KBD4X
 
+#define KALEIDOSCOPE_BOOTLOADER_FLIP_WORKAROUND 1
+
 #include <Arduino.h>
 #define HARDWARE_IMPLEMENTATION kaleidoscope::hardware::kbdfans::KBD4x
-#include "Kaleidoscope-HIDAdaptor-KeyboardioHID.h"
-
-#include "kaleidoscope/macro_helpers.h"
-#include "kaleidoscope/hardware/avr/pins_and_ports.h"
 
-#include "kaleidoscope/hardware/ATMegaKeyboard.h"
+#include "kaleidoscope/driver/keyscanner/ATMega.h"
+#include "kaleidoscope/driver/bootloader/avr/FLIP.h"
+#include "kaleidoscope/hardware/avr/AVRDevice.h"
 
 namespace kaleidoscope {
 namespace hardware {
 namespace kbdfans {
-class KBD4x: public kaleidoscope::hardware::ATMegaKeyboard {
+
+struct KBD4xDeviceProps : kaleidoscope::hardware::avr::AVRDeviceProps {
+  typedef struct KBD4xKeyScannerProps : public kaleidoscope::driver::keyscanner::ATMegaProps {
```

Comment:
`typedef struct...` could be replaced by `struct KeyScannerProps`. No need for a typedef here.